LayoutTests/fast/url/query-expected.txt

```diff
@@ -8,7 +8,7 @@ PASS canonicalize('http://www.example.com/?as?df') is 'http://www.example.com/?a
 PASS canonicalize('http://www.example.com/?\x02hello bye') is 'http://www.example.com/?%02hello%7F%20bye'
 PASS canonicalize('http://www.example.com/?%40%41123') is 'http://www.example.com/?%40%41123'
 PASS canonicalize('http://www.example.com/?q=你好') is 'http://www.example.com/?q=%26%2320320%3B%26%2322909%3B'
-PASS canonicalize('http://www.example.com/?q=\ud800\ud800') is 'http://www.example.com/?q=%26%2355296%3B%26%2355296%3B'
+PASS canonicalize('http://www.example.com/?q=\ud800\ud800') is 'http://www.example.com/?q=%26%2365533%3B%26%2365533%3B'
```

Comment:
This result currently matches Chrome, although not Firefox.

Is there any effect outside of URL handling, or is this the only scenario where TextCodec::getUnencodableReplacement is called?

Reply:
It would impact form submission in a similar way. Except that (should, didn't check in WebKit yet) uses Unencodablehandling::Entities. Surprisingly there's no test coverage for that though. Possibly because it's hard to get lone surrogates in the pipeline to begin with.

Reply:
QuestionMarks is dead code: #12304.

Reply:
I can't figure out how to make `<form>` hit this with a lone surrogate. And that's the only other scenario there is from a web platform perspective.

(I don't understand why TextCodecUTF8::encodeUTF8 takes this enum btw. It doesn't seem to do anything with it, but removing that also seems non-trivial.)